test/ssr.test.js

```diff
@@ -34,8 +34,9 @@ describe('ssr: true, target: server, prod mode', () => {
   })
 
   test('render', async () => {
-    const { body } = await get('/')
+    const { body, headers } = await get('/')
     expect(body).toContain('nuxt-color-mode-script')
+    expect(headers['content-security-policy']).toBeUndefined()
```

Comment:
Why is it expected that there is no CSP header when the target is `server`? Does `vue-renderer` set the CSP itself then? I'm using the server target and SSR and within development everything works fine, but when I deploy my project to production, the CSP hash seems to be missing from `script-src` again and the color-mode cannot be changed.

Reply:
This is default scenario which build.csp is not enabled in nuxt.config.js

Reply:
> when I deploy my project to production, the CSP hash seems to be missing from script-src again and the color-mode cannot be changed.

CSP hash should work same in dev and production, can you double check the nuxt.config and http response header of your page ?

Reply:
Hmm, the header only includes the always changing hash for nuxt state on production, but `sha256-SzzThFtAnNrq8hsItwHFrLjbWhI0pfrqgYInXTrgWRI=` only on development too. Currently, you can see it here yourself: https://alpha.maev.si/ which is running version 0.93.1 of https://github.com/maevsi/maevsi/. In the footer there are links to change the color mode, but the console shows a CSP error. Not so when starting the project using `yarn dev`. During `yarn build` [L57](https://github.com/nuxt-community/color-mode-module/blob/3512f15707b831e045eee396baf6b50d9d10664b/lib/module.js#L57) is ran across, but when `vue-renderer` calls [`vue-renderer:ssr:csp`](https://github.com/jamesgeorge007/nuxt.js/blob/8fd21d1ea2dda2863ceb4b39387290d0c6aa1066/packages/vue-renderer/src/renderer.js#L422) [L58-L61](https://github.com/nuxt-community/color-mode-module/blob/3512f15707b831e045eee396baf6b50d9d10664b/lib/module.js#L58-L61) is not run. Here you can see the csp configuration: https://github.com/maevsi/maevsi/blob/1c19ad303f8ce275e67dbe204adcfd1bc0ab66c9/nuxt/nuxt.config.js#L373

Btw, why is this plugin not registered by using `addPlugin` like here: https://github.com/nuxt-community/ackee-module/blob/fa6ac7d45dddbca7be5fea47294b9e5cb2206568/src/module.ts#L33

Reply:
I think this is because you’re using this module as buildModule which is only for dev and build instead of production running. @Atinux Should we recommend using this module as a normal module?

Reply:
Great point, and indeed that solved the issue!

Reply:
Indeed adding a note in the README when using CSP could be nice. PR welcome ❤️